### PR TITLE
ci: 3.13, attestations, and some modernization

### PR DESCRIPTION
**Pull request by henryiii**
This does some modernization, and adds 3.13 testing and most importantly, the beautiful 3.13 classifier. :)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,23 +70,27 @@ jobs:
...
         with:
           python-version: ${{ matrix.python-version }}
+          allow-prereleases: true
 
-      - name: Requirements check
-        run: python -m pip list
+      - uses: astral-sh/setup-uv@v3
+
+      - name: Install nox
+        run: uv tool install nox
```
Comment (Saransh-cpp) [inline .github/workflows/ci.yml:86]: Should we `uvx`/install `nox` using `uv` consistently across this file, or is there a reason to switch between these 2 in different jobs?

**Review by Saransh-cpp — APPROVED** [review]
Thanks for this, @henryiii! :)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (henryiii) [reply .github/workflows/ci.yml:86]: I think one just used it on more lines. We can normalize to either one if that's better.

Comment (Saransh-cpp) [reply .github/workflows/ci.yml:86]: Ah, I see, that makes sense. Leaving it as it is looks okay. Thanks!

**Comment by henryiii** [comment]
@Saransh-cpp , it would be good to get patch releases before 3.13 is released, on Tuesday!

**Comment by Saransh-cpp** [comment]
On it 🫡

**Pull request merged**